pkgs/development/pharo/default.nix

```diff
@@ -0,0 +1,100 @@
+{ lib
+, stdenv
+, fetchurl
+, unzip
+, cmake
+, SDL2
+, gcc
+, gnumake
+, libffi
+, libuuid
+, openssl
+, git
+, freetype
+, pixman
+, libgit2
+, runtimeShell
+, libpng
+, cairo
+}:
+let
+  pharo-sources = fetchurl {
+    # It is necessary to download from there instead of from the repository because that archive
+    # also contains artifacts necessary for the bootstrapping.
+    url = "https://files.pharo.org/vm/pharo-spur64-headless/Linux-x86_64/source/PharoVM-10.0.5-2757766-Linux-x86_64-c-src.zip";
+    sha256 = "08d04f6jc8kiwkx18ffig7wbjalqfryicxaw3am3kjsxsn2v19cb";
+  };
+in
+stdenv.mkDerivation {
+  pname = "pharo";
+  version = "10.0.5";
+  src = pharo-sources;
+
+  nativeBuildInputs = [
+    unzip
+    cmake
+    SDL2
+    gcc
+    gnumake
+    libffi
+    libuuid
+    openssl
+    git
+    freetype
+    pixman
+  ];
+
+  cmakeFlags = [
+    # Necessary to perform the bootstrapping without already having Pharo available.
+    "-DGENERATED_SOURCE_DIR=."
+    "-DGENERATE_SOURCES=OFF"
+    # Prevents CMake from trying to download stuff.
+    "-DBUILD_BUNDLE=OFF"
+  ];
+
+  installPhase = ''
+    cmake --build . --target=install
+    mkdir -p "$out/lib"
+    mkdir "$out/bin"
+    cp build/vm/*.so* "$out/lib/"
+    cp build/vm/pharo "$out/bin/.pharo"
+    cat >"$out/bin/pharo" <<EOF
```

Comment:
This should be possible using wrapProgram / makeWrapper. See the nixpkgs manual and other derivations for some info